src/components/AssetInput/AssetInput.tsx

```diff
@@ -0,0 +1,249 @@
+                    "tw-w-full tw-flex tw-flex-row tw-flex-wrap tw-border tw-rounded hover:tw-border-black-90 dark:hover:tw-border-black-40 tw-group focus-visible:tw-outline-none",
+                    size === "large" ? "tw-h-[11.5rem]" : "tw-h-14",
+                    isOpen || isFocusVisible
+                        ? "tw-border-black-90 dark:tw-border-black-10"
+                        : "tw-border-black-20 dark:tw-border-black-80",
+                    isFocusVisible && FOCUS_STYLE,
+                ])}
+            >
+                <AssetThumbnail asset={asset} size={size} isActive={isOpen || isFocusVisible} />
+                <span
+                    className={merge([
+                        size === "large" ? "tw-h-14" : "tw-h-full",
+                        "tw-py-3 tw-pr-3 tw-pl-4 tw-flex tw-flex-col tw-items-start tw-font-normal tw-flex-1",
+                    ])}
+                >
+                    <span
+                        className={merge([
+                            "tw-text-black-100 tw-text-s group-hover:tw-font-medium dark:tw-text-white",
+                            (isOpen || isFocusVisible) && "tw-font-medium",
+                        ])}
+                    >
+                        {asset.name}
+                    </span>
+                    <AssetSubline asset={asset} />
+                </span>
+                <span className={merge([size === "large" ? "tw-h-14" : "tw-h-full", "tw-p-3 tw-flex tw-items-center"])}>
+                    <span className={merge(["tw-transition-transform", isOpen && "tw-rotate-180"])}>
+                        <IconCaretDown size={IconSize.Size16} />
+                    </span>
+                </span>
+            </button>
+
+            <AnimatePresence>
+                {isOpen && (
+                    <motion.div
+                        className="tw-absolute tw-left-0 tw-w-full tw-overflow-hidden tw-box-border tw-p-0 tw-shadow-mid tw-list-none tw-m-0 tw-mt-2"
+                        key={`asset-input-menu-${menuId}`}
+                        initial={{ height: 0 }}
+                        animate={{ height: "auto" }}
+                        exit={{ height: 0 }}
+                        transition={{ ease: [0.04, 0.62, 0.23, 0.98] }}
+                    >
+                        <FocusScope restoreFocus>
+                            <div {...overlayProps} ref={overlayRef}>
+                                <DismissButton onDismiss={() => menuState.close()} />
+                                <ActionMenu menuBlocks={actions} focus={focusStrategy} />
+                                <DismissButton onDismiss={() => menuState.close()} />
+                            </div>
+                        </FocusScope>
+                    </motion.div>
+                )}
+            </AnimatePresence>
+        </div>
+    );
+};
+
+export const AssetInput: FC<AssetInputProps> = ({
+    asset,
+    actions = [],
+    size = "small",
```

Comment:
what about using an enum here, as we do it i.e. in the Dropdown with `DropdownSize.Small`?

Reply:
aece366